### PR TITLE
Refactor docker locations and README.

### PR DESCRIPTION
**Pull request by michaelkaye**
This addresses #3224 - we need to have the README in the same directory as the Dockerfile, but we currently have the main synapse readme in the root folder.

Moving the Dockerfile into the subdirectory will make it slightly harder (add `-f docker/Dockerfile` to the build command) but will make the updating of the readme on docker hub much simpler.

Also, by moving parts out of contrib that don't belong there, we can make it clearer which parts of the packaging we're supporting directly (and hopefully incorporate more contrib code, which depends on the image).

### CODE REVIEW

**Comment by michaelkaye** [comment]
Hmm. Do I need a `recursive-exclude docker` here to make travis-ci happy? I'm slightly confused why `contrib/docker` doesn't raise the same error.

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+## Run
+
+We do not currently offer a `latest` image, as this has somewhat undefined semantics.
+We instead release only tagged versions so upgrading between releases is entirely
+within your control.
+
+### Using docker-compose (easier)
+
+This image is designed to run either with an automatically generated configuration
+file or with a custom configuration that requires manual edition.
```
Comment (richvdh) [inline docker/README.md]: edition?

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+An easy way to make use of this image is via docker-compose, see the (https://github.com/matrix-org/synapse/tree/develop/contrib/docker)[contrib] section of the synapse project for examples.
```
Comment (richvdh) [inline docker/README.md]: You can't use a comma to join independent clauses. s/,/./

**Comment by richvdh** [comment]
Turns out that `contrib/*` is explicitly excluded from `check-manifest` by `setup.cfg`.

I'm not really sure why some things are excluded that way, and others via MANIFEST.in. I think I prefer the latter. I think you want `prune docker` there.

**Comment by turt2live** [comment]
@michaelkaye I believe this was superseded by https://github.com/matrix-org/synapse/pull/3644 - correct?

**Comment by dali99** [comment]
>This is based on the changes in #3253 but old branch is old, so tidying it up and re-submitting as a new, clean PR.

**Pull request closed without merging**